hw/drivers/sensors/bmp388/src/bmp388.c

```diff
@@ -0,0 +1,4138 @@
+* notice, this list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright
+* notice, this list of conditions and the following disclaimer in the
+* documentation and/or other materials provided with the distribution.
+*
+* Neither the name of the copyright holder nor the names of the
+* contributors may be used to endorse or promote products derived from
+* this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+* CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+* IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL COPYRIGHT HOLDER
+* OR CONTRIBUTORS BE LIABLE FOR ANY
+* DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+* OR CONSEQUENTIAL DAMAGES(INCLUDING, BUT NOT LIMITED TO,
+* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+* WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+* ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE
+*
+* The information provided is believed to be accurate and reliable.
+* The copyright holder assumes no responsibility
+* for the consequences of use
+* of such information nor for any infringement of patents or
+* other rights of third parties which may result from its use.
+* No license is granted by implication or otherwise under any patent or
+* patent rights of the copyright holder.
+*
+* File   bmp388.c
+* Date   10 May 2019
+* Version	1.0.2
+*
+*/
+
+#include <assert.h>
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+
+#include "os/mynewt.h"
+#if MYNEWT_VAL(BUS_DRIVER_PRESENT)
+#include "bus/drivers/i2c_common.h"
+#else
+#include "hal/hal_spi.h"
+#include "hal/hal_i2c.h"
+#include "i2cn/i2cn.h"
+#endif
+#include "sensor/sensor.h"
+#include "sensor/temperature.h"
+#include "sensor/pressure.h"
+#include "bmp388/bmp388.h"
+#include "bmp388_priv.h"
+#include "hal/hal_gpio.h"
+#include "modlog/modlog.h"
+#include "stats/stats.h"
+#include <syscfg/syscfg.h>
+
+#define COMPENSTATE_DEBUG 0
+#define FIFOPARSE_DEBUG 0
+#define CLEAR_INT_AFTER_ISR 0
+#define BMP388_MAX_STREAM_MS 200000
+#define BMP388_DEBUG 0
+/*
+ * Max time to wait for interrupt.
+ */
+#define BMP388_MAX_INT_WAIT (10 * OS_TICKS_PER_SEC)
+
+const struct bmp388_notif_cfg dflt_notif_cfg[] = {
+    {
+      .event     = SENSOR_EVENT_TYPE_WAKEUP,
+      .int_num   = 0,
+      .notif_src = BMP388_INT_DRDY_STATE,
+      .int_type   = BMP388_DRDY_INT
+    },
+    {
+      .event     = SENSOR_EVENT_TYPE_WAKEUP,
+      .int_num    = 0,
+      .notif_src  = BMP388_INT_FIFOWTM_STATE,
+      .int_type    = BMP388_FIFO_WTMK_INT
+    },
+    {
+      .event     = SENSOR_EVENT_TYPE_WAKEUP,
```

Comment:
Why are all events named the same ? Also, events are more for tracking physical events like double taps, single taps and not for FIFO handling. For FIFO handling you should just use  something that is used in `bma253` which handles interrupts using `int_enable` structure.

Reply:
the modification for this part is still on going, it will be fixed later.

Reply:
already fixed it now